x-pack/plugins/reporting/public/panel_actions/get_csv_panel_action.tsx

```diff
@@ -98,7 +97,20 @@ export class ReportingCsvPanelAction implements ActionDefinition<ActionContext>
     }
 
     const { embeddable } = context;
-    return embeddable.getInput().viewMode !== ViewMode.EDIT && embeddable.type === 'search';
+
+    if (embeddable.type !== 'search') {
+      return false;
+    }
+
+    const savedSearch = embeddable.getSavedSearch();
+    const query = savedSearch.searchSource.getField('query');
```

Comment:
is it not possible that searchsource contains the query of type sql but it gets dropped (ignored) in the actual embeddable?